test/extended/cli/builds.go

```diff
@@ -0,0 +1,316 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	t "github.com/onsi/gomega/types"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	bcOutputTemplate       = "{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}"
+	bcOutputToNameTemplate = "{{ .spec.output.to.name }}"
+	bcSourceTypeTemplate   = "{{ .spec.source.type }}"
+	testDockerFileContent  = "FROM image-registry.openshift-image-registry.svc:5000/openshift/tests:latest"
```

Comment:
We should be using images from here https://github.com/openshift/origin/blob/e87c32d71e0462e3c32ee96e4a198d087bc6f001/test/extended/util/image/image.go#L128 I think that `ShellImage()` should work here. The reason for that is that those images are then available for mirroring and thus for offline testing.